config/decoder.go

```diff
@@ -342,7 +347,8 @@ func (d *decoder) mapping(childKey string, value reflect.Value, def string) erro
 		for key := range v {
 			subKey := fmt.Sprintf("%v", key)
 			if subKey == "" {
-				return fmt.Errorf("empty key leads to ambiguity for path: %q", childKey)
+				// We can confuse an empty map key with a root element.
+				return addKeyToError(childKey, errors.New("empty map key is ambigious"))
```

Comment:
nit: is it worth adding another function, newErrorWithKey(msg string, key string), so we don't have to create error twice? A small performance win but not sure how much of a difference it makes. Will leave this to you.